datadog_checks_base/datadog_checks/base/checks/win/wmi/sampler.py

```diff
@@ -145,12 +144,13 @@ def __init__(
         self._timeout_duration = timeout_duration
 
         self._runSampleEvent = Event()
-        self._sampleComplete = Event()
-        self.setDaemon(True)
+        self._sampleCompleteEvent = Event()
 
-        self.start()
+        thread = Thread(target=self._query_sample_loop, name=class_name)
+        thread.daemon = True
+        thread.start()
```

Comment:
setDaemon  is an old API: https://docs.python.org/2/library/threading.html#threading.Thread.setDaemon